store/tikv/client.go

```diff
@@ -100,18 +100,20 @@ func (a *connArray) Init(addr string, security config.Security) error {
 			grpc_prometheus.StreamClientInterceptor,
 			grpc_opentracing.StreamClientInterceptor(),
 		)
-		conn, err := grpc.Dial(
+
+		ctx, cancel := goctx.WithTimeout(goctx.Background(), dialTimeout)
+		conn, err := grpc.DialContext(
+			ctx,
 			addr,
 			opt,
-			grpc.WithTimeout(dialTimeout),
 			grpc.WithInitialWindowSize(grpcInitialWindowSize),
 			grpc.WithInitialConnWindowSize(grpcInitialConnWindowSize),
 			grpc.WithUnaryInterceptor(unaryInterceptor),
 			grpc.WithStreamInterceptor(streamInterceptor),
 			grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(MaxCallMsgSize)),
 			grpc.WithDefaultCallOptions(grpc.MaxCallSendMsgSize(MaxSendMsgSize)),
 		)
-
+		cancel()
```

Comment:
Why call cancel() here?

Reply:
I think it is safe and to call `cancel` here, to release the resource as quick as possible.

Reply:
If grpc handle cancel signal right (i.e, never store the ctx in its inner state), this confirms to the Go idiom to release resource.
But I'm not sure of grpc implementation. So is there any example of this kind of usage on their canonical website of document? @wentaoxu